### PR TITLE
[MCPE] Best possible dimension handling possible.

### PR DESCRIPTION
**Pull request by colinrgodsey**
Just... let it happen, its fine.

### CODE REVIEW

**Comment by Shevchik** [comment]
As i already said - no. I can accept relying on bukkit to send chunk publisher, nut not the dimension switch.

**Comment by Remu1012** [comment]
why not it if works it works :D

**Comment by Shevchik** [comment]
Because just works is not good enough.

**Comment by colinrgodsey** [comment]
The ordering doesnt make sense if you trigger change dimension from the packet. I can add extra state handling to try and trigger some of these actions when it gets the change dimension packet, but its just going to make the code more brittle and stupid. this is the most correct implementation. its not that it "just works", its that it works correctly, and in a simple, straight-forward, and less error-prone way.

**Comment by colinrgodsey** [comment]
dimension switch driven by the packet now, uses the cached dimension set by the PEChunkPublisher to determine if it has to send fake chunks and lock
still works just as well 👍 :

File: src/protocolsupport/protocol/storage/netcache/MovementCache.java [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
 public class MovementCache {
 
 	protected static final double acceptableError = 0.1;
+	protected static final int peIncreasedLeniencyMillis = 1000;
 
 	protected double x;
 	protected double y;
 	protected double z;
 	protected int teleportConfirmId;
+	protected double peClientX;
```
Comment (Shevchik) [inline src/protocolsupport/protocol/storage/netcache/MovementCache.java]: Move it all back down

File: src/protocolsupport/protocol/pipeline/version/v_pe/PEPacketDecoder.java [hunk 2]
```diff
@@ -92,6 +92,14 @@ public int readPacketId(ByteBuf from) {
 		return sReadPacketId(from);
 	}
 
+	public static int sPeakPacketId(ByteBuf from) {
```
Comment (Shevchik) [inline src/protocolsupport/protocol/pipeline/version/v_pe/PEPacketDecoder.java]: peek

**Comment by Shevchik** [comment]
Now it needs the implementation of single send of LoginSuccess PLAYER_SPAWN, so we can actually lock queue using this packet, not the separate custom payload message.

**Comment by Shevchik** [comment]
The custom payload change still needs to be removed. Or updated to use pre13 converter logic for both client -> server and server -> client.

**Comment by Shevchik** [comment]
Soo, let's try this.

**Pull request merged**